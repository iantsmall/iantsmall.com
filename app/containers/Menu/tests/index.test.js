/**
 *
 * Tests for Menu
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */
import React from 'react';
import { cleanup } from 'react-testing-library';
import render from 'utils/testRenderWithContext';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { Menu } from '../index';

describe('<Menu />', () => {
  afterEach(cleanup);
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const props = {
      dispatch: jest.fn(),
      menu: {},
    };
    render(<Menu {...props} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the initial snapshot', () => {
    const props = {
      dispatch: jest.fn(),
      menu: {},
    };
    const {
      container: { firstChild },
    } = render(<Menu {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
});
