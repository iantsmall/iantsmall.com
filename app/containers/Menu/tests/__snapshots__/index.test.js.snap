// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Should render and match the initial snapshot 1`] = `
<nav
  class="navbar navbar-expand-md navbar-dark bg-dark sticky-top"
>
  <div
    class="container"
  >
    <span
      class="navbar-brand"
    >
      <span>
        Ian T. Small
      </span>
    </span>
    <button
      aria-label="Toggle navigation"
      class="navbar-toggler"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="justify-content-start navbar-collapse collapse"
    >
      <span
        class="navbar-text"
      >
        <span>
          Fullstack Software Engineer
        </span>
      </span>
    </div>
    <div
      class="justify-content-end navbar-collapse collapse"
    >
      <div
        class="navbar-nav"
      >
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rb-event-key="/resume/"
            href="/resume/"
          >
            <span>
              Resume
            </span>
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rb-event-key="#comingSoon"
            href="#comingSoon"
          >
            <span>
              Site Roadmap
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
